src/core/index.ts

```diff
@@ -0,0 +1,358 @@
+      }
+
+      const accountExists = await store.isExistingAccount(accountId)
+      if (!accountExists) {
+        throw new Error(`Failed to send message: Account doesn't exist: account=${accountId}`)
+      }
+
+      const url = `${sendMessageUrl}/accounts/${accountId}/messages`
+      return axios
+        .post(url, Buffer.from(JSON.stringify(message)), {
+          timeout: 10000,
+          headers: {
+            'Content-type': 'application/octet-stream'
+          }
+        })
+        .then(response => response.data)
+    },
+
+    creditSettlement: async (accountId, amount, settlementId = uuid()) => {
+      let details = `amountToCredit=${amount} account=${accountId} settlementId=${settlementId}`
+
+      if (amount.isZero()) {
+        return
+      }
+
+      if (!isValidAmount(amount)) {
+        return log(`Error: Failed to credit settlement, invalid amount: ${details}`)
+      }
+
+      if (!isSafeKey(accountId)) {
+        return log(`Error: Failed to credit settlement, invalid account: ${details}`)
+      }
+
+      const accountExists = await store.isExistingAccount(accountId)
+      if (!accountExists) {
+        return log(`Error: Failed to credit settlement, account doesn't exist: ${details}`)
+      }
+
+      // Load all uncredited settlement amounts from Redis
+      const uncreditedAmounts = await store
+        .loadAmountToCredit(accountId)
+        .then(amount => {
+          if (!isValidAmount(amount)) {
+            throw new Error('Invalid uncredited amounts, database may be corrupted')
+          }
+
+          return amount
+        })
+        .catch(err => {
+          log(`Error: Failed to load uncredited settlement amounts: account=${accountId}`, err)
+          return new BigNumber(0)
+        })
+
+      if (uncreditedAmounts.isGreaterThan(0)) {
+        log(`Loaded uncredited amount of ${uncreditedAmounts} to retry notifying connector`)
+      }
+
+      const amountToCredit = amount.plus(uncreditedAmounts)
+
+      // amountToCredit must be positive and finite due to validation on amount & uncreditedAmounts
+      // ...so this Quantity should always be valid
+      const scale = amountToCredit.decimalPlaces()
+      const quantityToCredit = {
+        scale,
+        amount: amountToCredit.shiftedBy(scale).toFixed(0) // `toFixed` always uses normal (not exponential) notation
+      }
+
+      const notifySettlement = () =>
+        axios({
+          method: 'POST',
+          url: `${creditSettlementUrl}/accounts/${accountId}/settlements`,
+          data: quantityToCredit,
+          timeout: 10000,
+          headers: {
+            'Idempotency-Key': settlementId
+          }
+        })
+
+      details = `amountToCredit=${amountToCredit} account=${accountId} settlementId=${settlementId}`
+      log(`Notifying connector to credit settlement: ${details}`)
+
+      const amountCredited = await retryRequest(notifySettlement)
```

Comment:
I am wondering now that maybe the retry logic attempt retry for a small number of times. Then if still unsuccessful pop onto some internal queue in the storage layer. Then that queue can be tried a periodic intervals by the settlement engine. 

That may be happening in the code but not clear as the PR is quite big so excuse me if that is happening already.

Reply:
`retryRequest` will retry the maximum number of attempts, and if that fails, it'll return that it credited 0. Later in that function, the leftover amount (in this case, the full amount) gets saved as an uncredited settlement to the store.

Upon the next incoming settlement, it loads these uncredited settlements from the store, adds them to the new incoming settlement, and sends a new request for that total amount.